### PR TITLE
[validator]: update for validator@12.0.0

### PR DESCRIPTION
**Pull request by MunifTanjim**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/validatorjs/validator.js/tree/12.0.0>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by PranavSenthilnathan** [comment]
Since this is a popular library and this PR contains some significant changes, can you please take a look @samuelneff @CodeAnimal @drinchev @babolivier @kukoo1 @oktapodia @MorpheusXAUT @TitaneBoy @zjy01 @nidzov @Raigen @todd @nrschultz @Thomas-B @Antoine38660 @smff @truongkhanhduy95 @emmanuelgautier @dancrumb @lalayueh @rohitsud @Torhal @tgfjt @chrootsu @IOAyman @louy @kacepe @deptno @builtinnya @qqilihq @keatz55

**Review by louy — APPROVED** [review]

File: types/sequelize/index.d.ts [hunk 1]
```diff
@@ -31,13 +31,12 @@
...
 // Based on original work by: samuelneff <https://github.com/samuelneff/sequelize-auto-ts/blob/master/lib/sequelize.d.ts>
 
-/// <reference types="validator" />
-
-
 import * as _ from "lodash";
 import Promise = require("bluebird");
 import * as cls from "continuation-local-storage"
 
+type ValidatorJS = typeof import('validator')
```
Comment (oktapodia) [inline types/sequelize/index.d.ts]: Prefer to keep the same import way as other [imports](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/40192/files#diff-0e3744815a33a1dfe3285ef4f51888e9R36)
```suggestion
import * as ValidatorJS from "validator";
```

File: types/sequelize/v3/index.d.ts [hunk 2]
```diff
@@ -10,12 +10,11 @@
 
 // Based on original work by: samuelneff <https://github.com/samuelneff/sequelize-auto-ts/blob/master/lib/sequelize.d.ts>
 
-/// <reference types="validator" />
-
-
 import * as _ from "lodash";
 import * as Promise from "bluebird";
 
+type ValidatorJS = typeof import('validator')
```
Comment (oktapodia) [inline types/sequelize/v3/index.d.ts]: Prefer to keep the same import way as other [imports](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/40192/files#diff-b6b16ead6d43beabdd6bd9ef300fbd17R14)
```suggestion
import * as ValidatorJS from "validator";
```

**Review by oktapodia — CHANGES_REQUESTED** [review]
Just a little change for the sequelize part.

I'm not sure this is the right place to update all the validators packages & the sequelize packages in the same place.

Might be better to update them in a PR, and the sequelize part in another one. I believe it does not impact these.

**Comment by MunifTanjim** [comment]
> Just a little change for the sequelize part.
> 
> I'm not sure this is the right place to update all the validators packages & the sequelize packages in the same place.
> 
> Might be better to update them in a PR, and the sequelize part in another one. I believe it does not impact these.

I can revert the sequelize changes, but that breaks the CI test. What can we do about that?

**Review by oktapodia — APPROVED** [review]
I see, approved then, can you just use the same import way as other libraries in the sequelize files (like suggested)?

**Comment by louy** [comment]
Line 1427 in validator/index seems to be causing issues. I didn’t know you could reimport export from within the same file

**Pull request merged**